packages/stylelint-plugin-dialtone/lib/rules/no-mixins.js

```diff
@@ -0,0 +1,52 @@
+const stylelint = require("stylelint");
+
+const {
+  createPlugin,
+  utils: { report, ruleMessages, validateOptions },
+} = stylelint;
+
+const ruleName = '@dialpad/stylelint-plugin-dialtone/no-mixins';
+
+const messages = ruleMessages(ruleName, {
+  noMixinsRejected: (selector) => `Please avoid using LESS mixins "${selector}"`,
+});
+
+const meta = {
+  url: "https://github.com/dialpad/dialtone/blob/staging/packages/stylelint-plugin-dialtone/docs/rules/no-mixins.md",
+};
+
+/** @type {import('stylelint').Rule} */
+const ruleFunction = (primary) => {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: primary,
+    });
+
+    if (!validOptions) return;
+
+    // This iterates through one selector at a time, so you don't have to worry about checking for nested selectors.
+    root.walkRules((ruleNode) => {
+      const { source } = ruleNode;
+
+      // any line that starts with a period and ends with a semicolon is a LESS mixin.
+      const regex = /\.[\S]+;/gm;
```

Comment:
most of those are for defining mixins, we just want to find mixin usage, which I think this will cover. Maybe we should add `d-` into the regex though so it only finds dialtone mixins.

Reply:
No, I think we want to remove all mixins usage, so I'll prefer to find all of them instead of just dialtone's

Reply:
that is true haha

Reply:
only one I see there that might not work is the `#outer.inner();` format, but I don't think I've ever even seen one used this way in product.